platform/ios/src/MGLMapView.mm

```diff
@@ -3824,10 +3824,12 @@ - (void)setShowsUserLocation:(BOOL)showsUserLocation
         if ([self.delegate respondsToSelector:@selector(mapView:viewForAnnotation:)])
         {
             userLocationAnnotationView = (MGLUserLocationAnnotationView *)[self.delegate mapView:self viewForAnnotation:self.userLocation];
-            if (userLocationAnnotationView)
+            if ( ! [userLocationAnnotationView isKindOfClass:MGLUserLocationAnnotationView.class])
             {
-                NSAssert([userLocationAnnotationView.class isSubclassOfClass:MGLUserLocationAnnotationView.class],
-                         @"User location annotation view must be a subclass of MGLUserLocationAnnotationView");
+                // Ignore the user-provided annotation if it is not a kind of
+                // MGLUserLocationAnnotationView (and use the default user
+                // location annotation instead).
```

Comment:
Added the log message in https://github.com/mapbox/mapbox-gl-native/pull/6163/commits/384274bc705eca374e99b966089e05f1ed0b88df#diff-20502b6bc02986a4398d0da84b2d4b94R3836.